### ISSUE TITLE
Support named parameters to processes and (sub-)workflows

### ISSUE THREAD

**Issue by nh13** [nextflow-io/nextflow#2067]
For processes that require more than just a few inputs, I would like to used named parameters to ensure I don't make a mistake in the order of the inputs.

**Comment by nh13** [comment]
Would still like this feature. Authoring complex pipelines is still quite a chore with many inputs.

**Comment by giusmar** [comment]
Hi,
I'va a similar problem.
I created a yml file in wich i set alla parameters that i can send to nextflow with -parameters-file

In this yml file i set parameters for reads for a RNASeq as:

```
reads:
- - name_one
  - ['/path/to/one','path/to/two']
- - name_two
  - ['/path/to/one','path/to/two']
```

But when I set ${reads[0]}, due the feature of nextflow that run the process not in order, that command can refer to first read of read_one than read_two. 

Can you help me? Please

**Comment by nh13** [comment]
@giusmar can you create a new issue, and if it's similar, link to this issue there? I'd like to keep this issue focused on what I feel to be a very important QOL feature missing from nextflow.

**Comment by giusmar** [comment]
yes sure, sorry :) i deleted the issue so you can have clean this page.
I don't know if u can delete also this message, let me know.
Bye :)

**Comment by bentsherman** [comment]
Appears to be a duplicate of #2257 . I will close this one since the other one is more detailed.
